test/functional-portable/corerp/noncloud/resources/application_test.go

```diff
@@ -116,7 +113,22 @@ func Test_ApplicationGraph(t *testing.T) {
 				// assert that the graph is as expected
 				expected := []*v20231001preview.ApplicationGraphResource{}
 				testutil.MustUnmarshalFromFile("corerp-resources-application-graph-out.json", &expected)
-				require.ElementsMatch(t, expected, res.Resources)
+
+				// For easier comparison, we sort the resources by name.
+				sort.Slice(res.Resources, func(i, j int) bool {
+					return *res.Resources[i].Name < *res.Resources[j].Name
+				})
+				sort.Slice(expected, func(i, j int) bool {
+					return *expected[i].Name < *expected[j].Name
+				})
+
+				if len(res.Resources) != len(expected) {
+					require.ElementsMatch(t, expected, res.Resources)
```

Comment:
Why is there a case where the expected may not have the same length as the actual?